gobblin-metrics-libs/gobblin-metrics/src/main/java/org/apache/gobblin/metrics/OpenTelemetryMetrics.java

```diff
@@ -60,9 +63,12 @@ protected MetricExporter initializeMetricExporter(State state) {
         "OpenTelemetry endpoint must be provided");
     OtlpHttpMetricExporterBuilder httpExporterBuilder = OtlpHttpMetricExporter.builder();
     httpExporterBuilder.setEndpoint(state.getProp(ConfigurationKeys.METRICS_REPORTING_OPENTELEMETRY_ENDPOINT));
-    if (state.contains(ConfigurationKeys.METRICS_REPORTING_OPENTELEMETRY_HEADER_KEY) && state.contains(ConfigurationKeys.METRICS_REPORTING_OPENTELEMETRY_HEADER_VALUE)) {
-      httpExporterBuilder.addHeader(state.getProp(ConfigurationKeys.METRICS_REPORTING_OPENTELEMETRY_HEADER_KEY),
-          state.getProp(ConfigurationKeys.METRICS_REPORTING_OPENTELEMETRY_HEADER_VALUE));
+
+    if (state.contains(ConfigurationKeys.METRICS_REPORTING_OPENTELEMETRY_HEADERS)) {
+      Map<String, String> headers = parseHttpHeaders(state.getProp(ConfigurationKeys.METRICS_REPORTING_OPENTELEMETRY_HEADERS));
```

Comment:
looks like this might pass in `null`, no?

e.g.:
```
Optional.ofNullable(state.getProp(...)).ifPresent(headersMap ->
  headersMap.entrySet().forEach(header ->
    heb.addHeader(...)
  ));
```

Reply:
since we check if the key exists, there wouldn't be a scenario where the map can be null because it uses the JSON parser. If the map isn't a well formed json object, such as empty string, it would fail loudly here during the parse step.

Reply:
ok, cool